cfe_internal/enterprise/federation/federation.cf

```diff
@@ -129,6 +129,46 @@ bundle agent config
       "I'm paused so won't do any import/dump";
 }
 
+bundle agent semanage_installed
+# @brief Install semanage utility if selinux enabled and
+# cfengine_mp_fr_dependencies_auto_install class is defined
+# if not defined then only warn
+{
+  vars:
+    "semanage_action"
+      string => ifelse( "default:_stdlib_path_exists_semanage", "fix", "default:cfengine_mp_fr_dependencies_auto_install", "fix", "warn" ),
+      comment => "We only want to use semanage if it's available, or if we have
+                  indicated it's ok to install it automatically. This variable
+                  is subsequently used by a commands and packages promises to
+                  warn or fix based.";
+    debian_6|debian_7|debian_8|ubuntu_12|ubuntu_14|ubuntu_16|rhel_5::
+      "semanage_package" string => "policycoreutils";
+    debian_9|debian_10|ubuntu_18|rhel_8::
+      "semanage_package" string => "policycoreutils-python-utils";
+    centos_6|centos_7::
```

Comment:
Isn't `redhat_6` defined on both CentOS and RHEL? Please make sure we cover both CentOS and RHEL here.

Reply:
Not the way I read sysinfo.c:OSClasses() but I just scanned it, will test. :+1:

Reply:
I don't think so. Only redhat and redhat_derived are defined on centos 7 for me:

```
[root@10-0-2-15-static ~]# cf-promises --show-classes | grep redhat
redhat                                                       inventory,attribute_name=none,source=agent,hardclass
redhat_derived                                               source=promise,inventory,attribute_name=none
[root@10-0-2-15-static ~]# cf-promises --show-classes | grep centos
centos                                                       inventory,attribute_name=none,source=agent,derived-from-file=/etc/os-release,hardclass
centos_7                                                     inventory,attribute_name=none,source=agent,derived-from=sys.flavor,hardclass
centos_7_7                                                   inventory,attribute_name=none,source=agent,hardclass
```

so I could add make sure that redhat_n and centos_n are all filled in above.